### PR TITLE
refactor: adjusting the position of ai module

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/event.lua [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local constants = {
+    CREATE_NEW_HTTP_ROUTER = 1,
+}
+
+local _M = {
+    constants = constants,
```
Comment (membphis) [inline apisix/event.lua:23]: Change the name to uppercase?

**Review by membphis — DISMISSED** [review]

File: apisix/event.lua [hunk 2]
```diff
@@ -0,0 +1,40 @@
+--
```
Comment (membphis) [inline apisix/event.lua:1]: And I think we can move `event` to `apisix.core`

Comment (membphis) [reply apisix/event.lua:1]: @spacewander what do you think?

File: apisix/event.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/event.lua:23]: how about `REBUILD_ROUTER`?

Comment (tzssangglass) [reply apisix/event.lua:23]: or `BUILD_ROUTER`?

File: apisix/plugins/ai.lua [hunk 3]
```diff
@@ -43,10 +45,23 @@ local route_lrucache = core.lrucache.new({
     count = 512
 })
 
-local _M = {}
+local schema = {
```
Comment (soulbird) [inline apisix/plugins/ai.lua:48]: `local schema = {}` ?

File: apisix/event.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/event.lua:23]: you mean change `constants` to `CONST`?

Comment (tzssangglass) [reply apisix/event.lua]: done

File: apisix/plugins/ai.lua (hunk 3 above)
Comment (tzssangglass) [reply apisix/plugins/ai.lua]: done

File: apisix/core/event.lua [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+local CONST = {
+    BUILD_ROUTER = 1,
+}
+
+local _M = {
+    CONST = CONST,
+}
+
+local events = {}
+
```
Comment (membphis) [inline apisix/core/event.lua:27]: style: pls use two blank lines between different functions

**Review by membphis — APPROVED** [review]

File: apisix/core/event.lua (hunk 4 above)
Comment (tzssangglass) [reply apisix/core/event.lua:27]: done

File: t/core/config.t [hunk 5]
```diff
@@ -38,7 +38,7 @@ __DATA__
 GET /t
 --- response_body
 etcd host: http://127.0.0.1:2379
-first plugin: "real-ip"
+first plugin: "ai"
```
Comment (membphis) [inline t/core/config.t:41]: we can change the priority of ai plugin
we should avoid to change this test case

Comment (tzssangglass) [reply t/core/config.t:41]: If it is not necessary, I am going to change it in the next PR

**Review by soulbird — APPROVED** [review]

**Pull request merged**